src/services/alpacaService.ts

```diff
@@ -44,7 +86,45 @@ export const closeOpenPosition = async (symbol: string, percentage: string) => {
   }
 };
 
-export const postOrder = async (
+/* Closes the position (sells 100%). */
+export const stopLossSellOrder = async (symbol: string) => {
+  if (
+    !symbol ||
+    typeof symbol !== "string" ||
+    symbol.length < 2 ||
+    symbol.length > 5
+  ) {
+    throw Error;
+  }
+
+  console.log(`Stop loss on ${symbol}`);
+  await deleteOrder(symbol);
+};
+
+/* This is triggered when price has went up with 10% or more.*/
+export const takeProfitSellOrder = async (symbol: string) => {
+  if (
+    !symbol ||
+    typeof symbol !== "string" ||
+    symbol.length < 2 ||
+    symbol.length > 5
+  ) {
+    throw Error;
+  }
+
+  const value = await getHoldingInTicker(symbol);
+  const body: BodyInit = JSON.stringify({
+    side: "sell",
+    symbol: symbol,
+    time_in_force: "day",
+    notional: value * 0.5, // sell 50%
+  });
+
+  console.log(`Take profit on ${symbol}`);
+  return postOrder(body);
+};
+
+export const postNewBuyOrder = async (
   ticker: string,
   type: Side,
```

Comment:
Do not have "type" as argument. Set is hardcoded into the body that gets passed to "postOrder", otherwise this method gives no extra value.